receiver/prometheusreceiver/metrics_receiver_test.go

```diff
@@ -1076,6 +1076,15 @@ rpc_duration_seconds_sum{foo="no_quantile"} 101
 rpc_duration_seconds_count{foo="no_quantile"} 55
 `
 
+var target4Page1 = `
```

Comment:
Note that in OpenMetrics, the following is valid:

```
# A counter named "foo".  In OpenMetrics, the counter metric point includes _total, but the TYPE comment does not 
# TYPE foo counter
foo_total 0
# A gauge named foo_total, which also has the same name on the metric point.
# TYPE foo_total gauge
foo_total 1
```
It is [discouraged](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#suffixes) to do so, though.

We have to choose to either check the normalized or the non-normalized name first, though.  If we check normalized first, we hit your issue.  If we check non-normalized first, we hit the issue above.

Reply:
looks like so, unless we are able to know the metric type, it looks like we can only choose one issue to fix..

Reply:
My assertion above is incorrect.